swandaskcluster/cluster.py

```diff
@@ -171,6 +173,17 @@ def security(cls):
             require_encryption=True,
         )
 
+    def close(self, timeout = None):
+        '''
+        Releases the port assigned to the scheduler.
+        '''
+
+        aw = super().close(timeout)
```

Comment:
Why this before? Is there a way it could raise an exception and then we never release it?

Reply:
So the wait it is implemented now, it's asynchronous because I don't `await` the result of the superclass' `close` here, meaning that the port is released independently of what happens during `close`. It is [here](https://github.com/dask/dask-labextension/blob/main/dask_labextension/clusterhandler.py#L24-L33) where the final `await` happens and, if there is an exception, it is raised as an HTTPError (and we see it from the UI).

However, one could argue that we could be releasing the port and maybe the scheduler process was not correctly shut down by `close`. So perhaps we could await inside a try catch, and only release the port if it went well? We'd need to make this a coroutine then, because the Dask lab extension awaits on our close.

Reply:
Discussed in person, agreed to go for the latter option.